### PR TITLE
add registry route to get key from username/dataset

### PR DESCRIPTION
**Pull request by joehand**
We need a route to lookup the dat key from username/dataset.

This is kind of clunky because I didn't know how this fit in the existing API methods (hence the `/registry` part). Also just realized you were using params for the other APIs so I can switch to that.

* Should we require generic authentication for this?

### CODE REVIEW

**Comment by karissa** [comment]
It can be `/api/v1/:username/:dataset`

File: server/api/index.js [hunk 1]
```diff
@@ -53,6 +54,27 @@ module.exports = function (router, db, ship) {
     }
   }
 
+  router.on('api/v1/registry/:username/:dataset', {
```
Comment (karissa) [inline server/api/index.js]: should start with /

**Comment by joehand** [comment]
> It can be `/api/v1/:username/:dataset`

I started with that but then username params was `params.model` because of the other router and I got confused. Routing confuses me.

**Comment by karissa** [comment]
No worries :) It should hit the /api/v1/:username/:dataset route because that is the only one that matches. There are no other routes that match it. However, if we had only /api/v1/:username, that would collide with /api/v1/:model .

**Comment by karissa** [comment]
ok I fixed this by using a different router module @joehand take a look then i'll merge

File: server/router.js [hunk 2]
```diff
@@ -47,68 +45,58 @@ module.exports = function (opts, db) {
...
 
-  router.on('/:username/:dataset', {
-    get: function (req, res, params) {
-      // TOOD: do it in one db query not two
-      db.models.users.get({username: params.username}, function (err, user) {
+  router.get('/:username/:dataset', function (req, res) {
+    // TOOD: do it in one db query not two
+    db.models.users.get({username: req.params.username}, function (err, user) {
+      if (err) return res.end(err.message)
+      if (!user.id) return sendSPA('/404', req, res)
```
Comment (joehand) [inline server/router.js]: This needs to be `var user = results[0]` or something since query will return array. 

Forgot to fix this last night.

**Comment by joehand** [comment]
@karissa just that one fix in the frontend part for `/username/dataset` but looks good!

File: server/router.js (hunk 2 above)
Comment (karissa) [inline server/router.js]: I removed that route for now, will re-add in another PR. This route is not an API route, it will be a route for displaying HTML.

**Pull request merged**